FWCore/Framework/src/Worker.cc

```diff
@@ -377,4 +378,49 @@ namespace edm {
       earlyDeleteHelper_->moduleRan(iEvent);
     }
   }
+
+  void Worker::runAcquire(EventPrincipal const& ep,
+                          EventSetup const& es,
+                          ParentContext const& parentContext,
+                          WaitingTaskWithArenaHolder& holder) {
+
+    ModuleContextSentry moduleContextSentry(&moduleCallingContext_, parentContext);
+    try {
+      convertException::wrap([&]()
+      {
+        this->implDoAcquire(ep, es, &moduleCallingContext_, holder);
```

Comment:
Should just before this line you call `ranAcquireWithoutException_= false`?

Reply:
This was being set to false in reset function so it gets reset every event, even the case of dealing with exceptions from prefetching.

    void reset() {
      cached_exception_ = std::exception_ptr();
      state_ = Ready;
      waitingTasks_.reset();
      workStarted_ = false;
      ranAcquireWithoutException_ = false;
      numberOfPathsLeftToRun_ = numberOfPathsOn_;
    }

Reply:
ranAcquireWithoutException_ is now set in different places after the commit which wraps a task to handle the externalWork exceptions.